api/register.go

```diff
@@ -0,0 +1,42 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/mhghw/fara-message/jwt"
+)
+
+type RegUser struct {
+	ID              string `json:"id"`
+	Username        string `json:"username"`
+	FirstName       string `json:"first_name"`
+	LastName        string `json:"last_name"`
+	Password        string `json:"password"`
+	ConfirmPassword string `json:"confirm_password"`
+	Gender          string `json:"gender"`
+	DateOfBirth     string `json:"date_of_birth"`
+}
+
+// handle func for register,receives a json string according to the RegUser struct and generate a JWT with ID parameter
+func Register(c *gin.Context) {
+	var requestBody RegUser
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+
+		log.Print(http.StatusBadRequest, err)
+	}
+	if requestBody.Password == requestBody.ConfirmPassword {
```

Comment:
Always handle your logic in a way that favorable condition returned at the end.
like what we do in error handling. we don't check if the err is nil! cause then we need to continue our function in the `if err == nil` block (we may do that in some rare conditions).
In here you can reverse your condition check, check if requestBody.Password is not equal to confirmPassword, then if it was not, return the password does not match error!